### PR TITLE
Minor code cleanup

### PR DESCRIPTION
**Pull request by tangjiangling**
- Delete unused `HiveViewNotSupportedException`
- Rename field `allowedExceptions` to `stopOnExceptions`

## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Code cleanup.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

N/A.

> How would you describe this change to a non-technical end user or system administrator?

N/A.

## Documentation

( ) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java [hunk 1]
```diff
@@ -347,7 +346,7 @@ public Optional<Table> getTable(HiveIdentity identity, String databaseName, Stri
     {
         try {
             return retry()
-                    .stopOn(NoSuchObjectException.class, HiveViewNotSupportedException.class)
```
Comment (findinpath) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java:350]: This cleanup makes sense to me.
There are no Trino related calls which could throw `HiveViewNotSupportedException` in the callables wrapped by retry.

Please split the changes from this file in a separate commit.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/RetryDriver.java [hunk 2]
```diff
@@ -50,15 +50,15 @@ private RetryDriver(
...
+            List<Class<? extends Exception>> retryStoppedExceptions,
             Optional<Runnable> retryRunnable)
     {
         this.maxAttempts = maxAttempts;
         this.minSleepTime = minSleepTime;
         this.maxSleepTime = maxSleepTime;
         this.scaleFactor = scaleFactor;
         this.maxRetryTime = maxRetryTime;
-        this.allowedExceptions = allowedExceptions;
+        this.retryStoppedExceptions = retryStoppedExceptions;
```
Comment (findinpath) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/RetryDriver.java:61]: I would argue that `stopOnExceptions`  brings more meaning compared to `retryStoppedExceptions`.
See also the method `io.trino.plugin.hive.util.RetryDriver#stopOn`

**Comment by tangjiangling** [comment]
AC @findinpath PTAL

**Comment by findinpath** [comment]
nit:
Replace commit messages with the following ones:

- Delete unused HiveViewNotSupportedException
- Rename field `allowedExceptions` to `stopOnExceptions`

The messages are more concise and there is no confusion about what they are doing.

**Comment by tangjiangling** [comment]
AC

**Comment by findinpath** [comment]
@findepi  PTAL

**Comment by tangjiangling** [comment]
Ping @findepi

**Pull request merged**